pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/BaseSegmentFetcher.java

```diff
@@ -109,6 +111,46 @@ public File fetchUntarSegmentToLocalStreamed(URI uri, File dest, long rateLimit,
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * @param segmentName
+   * @param uriSupplier the supplier to the list of segment download uris.
+   * @param dest        The destination to put the downloaded segment.
+   * @return true if and only if the segment fetch is successful. This method keeps retrying (with exponential backoff)
+   * of the following steps until segment download is successful or the retry limit is reached whichever comes first 1)
+   * Find servers hosting the segment in ONLINE state from the External View of the table. 2) Shuffle the list of
```

Comment:
Need to update javadoc. Since we are using a Supplier the logic to get the URIs is configurable.

Reply:
done.